src/main/java/jrobo/command/InputParser.java

```diff
@@ -0,0 +1,87 @@
+package jrobo.command;
+
+import jrobo.exception.InvalidFormatException;
+
+public class InputParser {
+    protected String[] args;
+    protected static String[] validCommands = {"mark", "m", "unmark", "um", "todo", "t", "deadline", "d", "event", "e", "list", "ls", "bye", "exit", "quit", "q"};
+
+    public InputParser(String input) {
+        this.args = input.split(" ");
+    }
+
+    public String getPrefix() {
+        for (String s : validCommands) {
+            String command = args[0];
+            if (command.equalsIgnoreCase(s)) {
+                return command.toLowerCase();
+            }
+        }
+        return null;
+    }
+
+    public String getBody() {
+        int suffixIndex = findSuffixIndex();
+
+        StringBuilder description = new StringBuilder();
+        int n = suffixIndex == -1 ? args.length : suffixIndex;
+        for (int i = 1; i < n; i++) {
+            description.append(" ").append(args[i]);
+        }
+        return description.toString();
+    }
+
+    public String getSuffix() {
+        int suffixIndex = findSuffixIndex();
+        if (suffixIndex == -1) {
```

Comment:
Avoid magic numbers. Consider declaring a variable (e.g. `INVALID_INDEX`) for the integer `-1` instead.